<?xml version="1.0"?>
<robot name="motor" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <link name="base_footprint"/>

  <link name="base_link">
      <visual>
        <origin rpy="0 0 0 " xyz="0 0 0.005"/>
        <geometry>
            <box size="0.4 0.4 0.01"/>
        </geometry>
      </visual>
      <collision>
        <origin rpy="0 0 0 " xyz="0 0 0.005"/>
        <geometry>
          <box size="0.4 0.4 0.01"/>
        </geometry>
      </collision>
      <inertial>
        <origin rpy="0 0 0 " xyz="0 0 0.005"/>
        <geometry>
          <box size="0.4 0.4 0.01"/>
        </geometry>
        <mass value="100"/>
        <inertia ixx="1.0265" ixy="0" ixz="0" iyy="1.3944999999999999" iyz="0" izz="2.1320000000000001"/>
      </inertial>
  </link>

  <joint name="base_joint" type="fixed">
      <parent link="base_footprint"/>
      <child link="base_link"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>

  <link name="wheel_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <cylinder length="0.05" radius="0.11"/>
      </geometry>
    </visual>
    <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.05" radius="0.11"/>
        </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
          <cylinder length="0.05" radius="0.11"/>
      </geometry>
        <mass value="3.5"/>
        <inertia ixx="0.0267" ixy="0" ixz="0" iyy="0.0267" iyz="0" izz="0.0506"/>
    </inertial>
  </link>

  <xacro:include filename="$(find gazebo_ros_motors)/xacro/dc_motor.xacro"/>
  
  <xacro:dc_motor motor_name="dc_motor" parent_link="base_link" child_link="wheel_link">
    <xacro:property name="params_yaml" value="$(find gazebo_ros_motors)/params/dc_motor.yaml"/>
    <origin xyz="0 0 0.2" rpy="0 0 0"/>
  </xacro:dc_motor>

  <!-- <xacro:include filename="$(find gazebo_ros_motors)/xacro/joint_motor.xacro"/>
  <xacro:joint_motor motor_name="ode_motor" parent_link="base_link" child_link="wheel_link">
    <xacro:property name="params_yaml" value="$(find gazebo_ros_motors)/params/joint_motor.yaml"/>
    <origin xyz="0 0 0.2" rpy="0 0 0"/>
  </xacro:joint_motor> -->

</robot>
